client/interface.go

```diff
@@ -0,0 +1,33 @@
+package client
+
+import (
+	"encoding"
+
+	"github.com/libp2p/go-libp2p-core/event"
+)
+
+// Client represents the drand Client interface.
+type Client interface {
+	// TextMarshaler (the 'MarshalText' method) allows serialization of the client
+	// state for subsequent restoration. The marshalled value can be passed to `New`
+	// to restore Client state.
+	encoding.TextMarshaler
+
+	// Get returns a the randomness at `round` or an error over a result channel.
+	// One result will be returned, with exactly one of Err or Data populated.
+	func Get(ctx context.Context, round uint64) <-chan Result
+
+	// RoundAt will return the most recent round of randomness that will be available
+	// at time for the current client.
+	func RoundAt(time time.Time) uint64
+}
+
+// Result represents the randomness for a single drand round.
+type Result struct {
+	Err error
+	Round uint64
+	Data []byte
```

Comment:
I would prefer to call this field `Signature` - data is too generic imo (is the data the thing I can use for randomness ? is it the signature?)

Reply:
per https://github.com/drand/drand/blob/master/protobuf/drand/api.pb.go#L76-L81

should we return the `Signature` directly, or should we hash it?

Reply:
We could directly return the randomness that I understand you already verify it. But the hash function depends on the application, there's no good or bad answers so it's as well ok if we let the application hash the signature the way it wants.

Reply:
I'd tend to lean to the side of "make it hard for the consumer to be able to shoot themselves in the foot". Providing the thing we expect to be uniform randomness, and not other things that could be confused with it seems like the cleanest way to do that.

Reply:
So let's add an explicit `Randomness` field which is the `sha256` of the randomness then :)